src/main/java/io/micronaut/build/services/DockerService.java

```diff
@@ -104,6 +114,36 @@ public void onNext(BuildResponseItem item) {
                 .awaitImageId();
     }
 
+    /**
+     * Creates a container based on a given image, and runs it.
+     * @param imageId the image to use
+     */
+    public void runPrivilegedImageAndWait(String imageId, String... binds) throws IOException {
+        try (CreateContainerCmd create = dockerClient.createContainerCmd(imageId)) {
+            HostConfig hostConfig = create.getHostConfig();
+            if (hostConfig == null) {
+                LOG.error("HostConfig is null");
```

Comment:
Can this message be improved? Perhaps a hint to the user about what happened or what to do?

Reply:
This should never be null (though I guess it _can_ be), and if it is, it's kinda catastrophic...

I've switched to throwing an exception here, recommending people check their docker installation 🤔